backend/signing/configuration.go

```diff
@@ -240,3 +240,22 @@ func (configuration *Configuration) String() string {
 	}
 	return fmt.Sprintf("single sig, scriptType: %s", configuration.scriptType)
 }
+
+// Configurations is a collection of configurations.
+type Configurations []*Configuration
+
+// Hash returns a hash of all configurations in hex format. It is defined as
+// `sha256(<32 bytes hash 1>|<32 bytes hash 2>|...)`.
+func (configs Configurations) Hash() string {
+	h := sha256.New()
+	for _, cfg := range configs {
```

Comment:
This makes the resulting hash order-dependent: `[]Configurations{a, b}` and `[]Configurations{b, a}` is the same functionality during runtime, right? But the hash is different.

It just seems a bit fragile.

Reply:
Good point. I commented that the slice is unordered and now sort the individual hashes before hashing again. Added test to make sure the ordering does not matter.